internal/xcuit/config.go

```diff
@@ -0,0 +1,136 @@
+package xcuit
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/saucelabs/saucectl/internal/config"
+	"gopkg.in/yaml.v2"
+)
+
+var supportedDeviceTypes = []string{"ANY", "PHONE", "TABLET"}
+
+// Project represents the xcuit project configuration.
+type Project struct {
+	config.TypeDef `yaml:",inline"`
+	ConfigFilePath string             `yaml:"-" json:"-"`
+	Sauce          config.SauceConfig `yaml:"sauce,omitempty" json:"sauce"`
+	Xcuit          Xcuit              `yaml:"xcuit,omitempty" json:"xcuit"`
+	Suites         []Suite            `yaml:"suites,omitempty" json:"suites"`
+	Artifacts      config.Artifacts   `yaml:"artifacts,omitempty" json:"artifacts"`
+}
+
+// Xcuit represents xcuit apps configuration.
+type Xcuit struct {
+	App     string `yaml:"app,omitempty" json:"app"`
+	TestApp string `yaml:"testApp,omitempty" json:"testApp"`
+}
+
+// TestOption represents the xcuit test filter options configuration.
+type TestOption struct {
+	Class []string `yaml:"class,omitempty" json:"class"`
+}
+
+// Suite represents the xcuit test suite configuration.
+type Suite struct {
+	Name        string     `yaml:"name,omitempty" json:"name"`
+	Devices     []Device   `yaml:"devices,omitempty" json:"devices"`
```

Comment:
Shouldn't we reuse `config.Device` here ?

Reply:
Good idea! I reused `config.Device`.